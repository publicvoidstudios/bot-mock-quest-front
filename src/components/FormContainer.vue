<template>
  <div class="fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 bg-secondary flex flex-col items-center justify-center rounded-lg shadow-lg p-4 w-[100%] sm:w-auto">
    <slot></slot>
  </div>
</template>