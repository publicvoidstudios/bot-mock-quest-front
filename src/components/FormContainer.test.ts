import { mount } from '@vue/test-utils';
import {describe, it, expect} from "vitest";
import FormContainer from './FormContainer.vue';

describe('FormContainer.vue', () => {
    const wrapper = mount(FormContainer);

    it('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('has all necessary classes to be fixed in middle of a screen', () => {
        expect(wrapper.classes()).toContain('fixed');
        expect(wrapper.classes()).toContain('top-1/2');
        expect(wrapper.classes()).toContain('left-1/2');
        expect(wrapper.classes()).toContain('-translate-x-1/2');
        expect(wrapper.classes()).toContain('-translate-y-1/2');
    });
});