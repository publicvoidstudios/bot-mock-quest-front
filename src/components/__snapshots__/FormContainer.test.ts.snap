// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormContainer.vue > renders correctly 1`] = `
<div
  class="fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 bg-secondary flex flex-col items-center justify-center rounded-lg shadow-lg p-4 w-[100%] sm:w-auto"
>
  
  
</div>
`;
